mex/common/identity/memory.py

```diff
@@ -1,3 +1,5 @@
+import hashlib
```

Comment:
i'd rename this to something like `HashingIdentityProvider` and trash the "database" attribute, should be way cleaner (as discussed)

Reply:
Unfortunately, this is not possible. To generate identifiers, we `hadPrimarySource` and `identifierInPrimarySource`. These parameters are optional in the current `fetch`-method implementation. Additionaly, the `fetch` method supports getting identifiers based on their stableTargetId, which is impossible with the hash-based method and no database. Hence, I will leave the database in there and will also keep the name. 
Is that alright with you?

Reply:
yep, convinced.